user/caribou/module/src/DSO9254AEventConverter.cc

```diff
@@ -0,0 +1,448 @@
+#include "CaribouEvent2StdEventConverter.hh"
+#include "utils/log.hpp"
+#include "TFile.h"
+#include "TH1D.h"
+#include "TGraph.h"
+#include "time.h"
+
+using namespace eudaq;
+
+namespace{
+  auto dummy0 = eudaq::Factory<eudaq::StdEventConverter>::
+    Register<DSO9254AEvent2StdEventConverter>(DSO9254AEvent2StdEventConverter::m_id_factory);
+}
+
+
+bool    DSO9254AEvent2StdEventConverter::m_configured(0);
+int DSO9254AEvent2StdEventConverter::m_channels(0);
+int DSO9254AEvent2StdEventConverter::m_digital(1);
+bool DSO9254AEvent2StdEventConverter::m_generateRoot(1);
+
+bool DSO9254AEvent2StdEventConverter::Converting(eudaq::EventSPC d1, eudaq::StandardEventSP d2, eudaq::ConfigurationSPC conf) const{
+
+
+  auto ev = std::dynamic_pointer_cast<const eudaq::RawEvent>(d1);
+  // No event
+  if(!ev) {
+    return false;
+  }
+
+  // load parameters from config file
+  std::ofstream outfileTimestamps;
+  if (!m_configured) {
+    // generate rootfile to write waveforms as TH1D  - RECREATE it here and append later
+    TFile *histoFile = nullptr;
+    if (m_generateRoot) {
+      histoFile =
+          new TFile(Form("waveforms_run%i.root", ev->GetRunN()), "RECREATE");
+      if (!histoFile) {
+        EUDAQ_ERROR(to_string(histoFile->GetName()) + " can not be opened");
+        return false;
+      }
+      histoFile->Close();
+    }
+    // check for tags that are specific for a oszi
+    auto tags = d1->GetTags();
+    m_digital = 0;
```

Comment:
You only accept the values from the configuration file if no config is dumped in the data (older samples). If you now put a 1 in the config for whatever reason, but the config in the data does not have a digital channel this will stay 1, that's why it is only read from the file later and set to zero here - at least that was my idea :)

Reply:
That means that it defaults to values inferred from data, but if one gives the parameters, these are prioritized?

Reply:
nope, whatever is in the data is the highest priority. User input is only used if nothing is in the data